components/builder-depot/src/server.rs

```diff
@@ -1779,12 +1691,50 @@ fn show_package(req: &mut Request) -> IronResult<Response> {
                 session_id,
                 &ident.get_origin(),
             ));
-            request.set_ident(ident);
+            request.set_ident(ident.clone());
+            let u = helpers::extract_query_value("upstream", req);
+
+            // If the "upstream" query parameter is not passed, then we can assume it's a normal
+            // API request and we should check for an update in the background.
```

Comment:
I'm not sure I understand the need for the upstream parameter..

Reply:
The need for it revolved around the UX choice I made for downloading a package via `hab pkg install`.  Since I went with synchronously checking the upstream depot in-band, I needed a way to differentiate between that kind of request and a normal API call to e.g. `/v1/depot/pkgs/core/redis/1.2.3/20180101123456`